File: CIVITAS/src/civitas/Impuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class Impuesto extends Casilla{
    
    private float importe;
    
    Impuesto (float cantidad , String nombre){
     super(nombre);
     this.importe = cantidad;

    }
       
       
    @Override
    public void recibeJugador(int iactual , ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(iactual, todos)){
            this.informe(iactual, todos);
            todos.get(iactual).pagaImpuesto(this.importe);
        }
    }
    
    @Override
    public String toString() {
        String result = "\nCasilla{\n" + "nombre=" + this.getNombre();
        if(this.importe != -1.0){
            result += "\nImporte = "+ this.importe;
        }
      
        
        result += "\n}";
        return result;
    }

}

File: CIVITASRUBY/lib/impuesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  
  require_relative 'casilla'
  class Impuesto < Casilla
     def initialize(carcel,cantidad,nombre)
      super(nombre,carcel)
      @importe = cantidad
    end
    
    def recibe_jugador(actual,todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos[actual].paga_impuesto(@importe)
      end
    end
    
  
    
    def to_s
      result = "Casilla: \nnombre= #{@nombre} \n tipo= #{@tipo} "
      
      if @importe != -1
        result +="\n"
        result += "\nimporte = #{@importe}"
      end
      result +="\n"
      
      result
      
    end
  end
end

